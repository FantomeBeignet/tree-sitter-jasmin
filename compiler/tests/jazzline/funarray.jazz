fn fill (reg u64[4] t) -> reg u64[4]
requires #[prover=smt] { 0 == t[0] }
ensures #[prover=smt] { result.0[0] == t[0] }
{
  inline int i;
  for i = 0 to 4 {
    t[i] = i;
  }
  return t;
}

fn f () -> reg u64 {
  reg u64[4] t;
  /* Remark the array_copy introduced after inlining does not make sens ...
     A solution will be to do a "After variable renaming to remove copies"
  */
  t = fill(t);

  reg u64 r;
  inline int i;

  r = 0;
  for i = 0 to 4 {
    r += t[i];
  }
  return r;

}