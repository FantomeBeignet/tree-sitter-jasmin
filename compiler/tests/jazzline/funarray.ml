open Common

let () =
  parse_and_print "funarray.jazz"  "fill";
  parse_and_print "funarray.jazz"  "f"
