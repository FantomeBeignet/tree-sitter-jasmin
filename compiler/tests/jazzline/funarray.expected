proc fill(t_3_184@uint64, t_2_183@uint64, t_1_182@uint64, t_0_181@uint64,
          t_0_161@uint64, t_1_162@uint64, t_2_163@uint64, t_3_164@uint64) =
{
 true && ((const 64 (0)) = t_0_161@uint64)
}
mov t_0_181@uint64 (0)@uint64;
mov t_1_182@uint64 (1)@uint64;
mov t_2_183@uint64 (2)@uint64;
mov t_3_184@uint64 (3)@uint64;
{
 true && (t_0_181@uint64 = t_0_161@uint64)
}
 proc f(t_3_279@uint64, t_2_278@uint64, t_1_277@uint64, t_0_276@uint64,
       r_284@uint64) =
{
 true && true
}
mov t_0_272@uint64 (0)@uint64;
mov t_1_273@uint64 (0)@uint64;
mov t_2_274@uint64 (0)@uint64;
mov t_3_275@uint64 (0)@uint64;
assert true && ((const 64 (0)) = t_0_272@uint64);
assume true && (t_0_276@uint64 = t_0_272@uint64);
mov r_280@uint64 (0)@uint64;
adds NONE_____285@uint1 r_281@uint64 r_280@uint64 t_0_276@uint64;
adds NONE_____286@uint1 r_282@uint64 r_281@uint64 t_1_277@uint64;
adds NONE_____287@uint1 r_283@uint64 r_282@uint64 t_2_278@uint64;
adds NONE_____288@uint1 r_284@uint64 r_283@uint64 t_3_279@uint64;
{
 true && true
}
 